modules/queue/queue_channel.go

```diff
@@ -75,6 +81,39 @@ func (q *ChannelQueue) Push(data Data) error {
 	return nil
 }
 
+// Shutdown processing from this queue
+func (q *ChannelQueue) Shutdown() {
+	log.Trace("ChannelQueue: %s Shutting down", q.name)
+	select {
```

Comment:
invoke `q.shudownCancel()` before waiting for the `q.shutdownCtx.Done`. And we could have a timeout wait on the same select and return directly if timeout

Reply:
If you do that multiple calls to shutdown will invoke flushers. If you have this select block you only create a terminate ctx flusher on the first shutdown.

Reply:
Ah I think you've misunderstood what shutdownCancel is - it's the CancelFunc for the shutdownCtx. I've renamed things to make them a little clearer.